### PR TITLE
[Deeplink] Change Deeplink parameter for ride type into an enum.

### PR DESCRIPTION
**Pull request by slowhill**
Restrict the ride type that can be passed into the deeplink constructor.
Update the enum names for the Lyft Button to match naming standards.

### CODE REVIEW

**Comment by weefbellington** [comment]
👍

File: lyft-button/src/main/java/com/lyft/lyftbutton/RideTypeEnum.java [hunk 1]
```diff
@@ -18,12 +18,12 @@ public String toString() {
     }
 
     public String getDisplayName() {
-        if (this == LINE) {
-            return "Lyft Line";
-        } else if (this == CLASSIC) {
+        if (this == SHARED) {
```
Comment (pohh) [inline lyft-button/src/main/java/com/lyft/lyftbutton/RideTypeEnum.java]: These need to be localized.

File: deeplink/src/main/java/com/lyft/deeplink/DeepLink.java [hunk 2]
```diff
@@ -63,7 +63,7 @@ public static boolean launchLyftApp(Context context, DeepLinkParams deepLinkPara
     static String createDeepLinkString(DeepLinkParams deepLinkParams) {
         StringBuilder sb = new StringBuilder();
         sb.append("lyft://ridetype?id=");
-        sb.append(deepLinkParams.getRideType());
+        sb.append(deepLinkParams.getRideType().getRideTypeKey());
```
Comment (pohh) [inline deeplink/src/main/java/com/lyft/deeplink/DeepLink.java]: This is a null pointer exception according to the current implementation of `DeepLinkParams.java`.

Comment (pohh) [reply deeplink/src/main/java/com/lyft/deeplink/DeepLink.java]: https://github.com/lyft/lyft-android-sdk/pull/13/files#diff-6edc060b52e27034e818eed5d40e3371R42

File: lyft-button/src/main/java/com/lyft/lyftbutton/RideTypeEnum.java (hunk 1 above)
Comment (pohh) [reply lyft-button/src/main/java/com/lyft/lyftbutton/RideTypeEnum.java]: nit: You could also make this a property in the enum.

File: deeplink/src/main/java/com/lyft/deeplink/DeepLink.java (hunk 2 above)
Comment (slowhill) [reply deeplink/src/main/java/com/lyft/deeplink/DeepLink.java]: Fixed the NPE, take a look when you get the chance and let me know if it's good to go :+1:

**Pull request merged**